lib/src/features/city_details/data/models/city_model.dart

```diff
@@ -0,0 +1,49 @@
+
+import 'package:emotion/src/features/city_details/data/models/city_provider_model.dart';
+import 'package:emotion/src/features/city_details/data/models/country_model.dart';
+import 'package:emotion/src/features/city_details/domain/entities/city.dart';
+
+class CityModel extends City {
```

Comment:
What's the idea behind dividing these classes into `entities` and `models`? From what I understand, the only difference between models and entities is that models include `fromJson` and `toJson` methods. They even share the same constructors.

Reply:
The idea was to split "scheme" - so called entity at domain layer. Models are implemented in the data layer which is used and interacted by app. Well for now I think it would be good to stick with just models (entities takes time and in this project it might be just overkill).